File: be-engine-v1/be-socket/src/main/java/com/cmcorg20230301/be/engine/socket/model/enums/SysSocketOnlineTypeEnum.java
package com.cmcorg20230301.be.engine.socket.model.enums;

import com.baomidou.mybatisplus.annotation.EnumValue;
import com.fasterxml.jackson.annotation.JsonValue;
import lombok.AllArgsConstructor;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * socket 在线状态
 */
@AllArgsConstructor
@Getter
public enum SysSocketOnlineTypeEnum {

    ONLINE(101), // 在线

    HIDDEN(201), // 隐身

    PING_TEST(100001), // ping测试，用于：获取延迟最低的 socket服务器

    ;

    @EnumValue
    @JsonValue
    private final int code;

    @NotNull
    public static SysSocketOnlineTypeEnum getByCode(Integer code) {

        if (code == null) {
            return ONLINE;
        }

        for (SysSocketOnlineTypeEnum item : SysSocketOnlineTypeEnum.values()) {

            if (item.getCode() == code) {

                return item;

            }

        }

        return ONLINE;

    }

}

File: fe-antd-v1/src/model/enum/SysSocketOnlineTypeEnum.ts
import {IEnum} from "@/model/enum/CommonEnum";
import {ProSchemaValueEnumType} from "@ant-design/pro-components";

export interface ISysSocketOnlineTypeEnum {

    ONLINE: IEnum<number>,

    HIDDEN: IEnum<number>,

    PING_TEST: IEnum<number>,

}

// socket 在线状态，枚举类
export const SysSocketOnlineTypeEnum: ISysSocketOnlineTypeEnum = {

    ONLINE: {
        code: 101,
        name: '在线',
    },

    HIDDEN: {
        code: 201,
        name: '隐身',
    },

    PING_TEST: {
        code: 100001,
        name: 'ping测试',
    },

}

export const SysSocketOnlineTypeEnumDict = new Map<number, ProSchemaValueEnumType>();

Object.keys(SysSocketOnlineTypeEnum).forEach(key => {

    const item = SysSocketOnlineTypeEnum[key];

    SysSocketOnlineTypeEnumDict.set(item.code as number, {text: item.name})

})
